tailscale/client.go

```diff
@@ -415,28 +478,53 @@ func WithETag(etag string) SetACLOption {
 	}
 }
 
-// SetACL sets the ACL for the given tailnet.
-func (c *Client) SetACL(ctx context.Context, acl ACL, opts ...SetACLOption) error {
+// SetACL sets the ACL for the given tailnet. `acl` can either be an [ACL],
+// or a HuJSON string.
+func (c *Client) SetACL(ctx context.Context, acl any, opts ...SetACLOption) error {
```

Comment:
I suppose the any here could have a stricter type constraints like:

```
type Pol interface {
    ACL | string
}

func (c *Client) SetACL[Pol T](ctx context.Context, acl T, opts ...SetACLOption) error {
```

But I am not convinced it would be more readable...

Reply:
That would work if `SetACL` was a regular function, but I don't think you can do that on a method.